spyder/plugins/editor/widgets/completion.py

```diff
@@ -248,16 +231,37 @@ def get_html_item_representation(self, item_completion, item_type,
                                      height=DEFAULT_COMPLETION_ITEM_HEIGHT,
                                      width=DEFAULT_COMPLETION_ITEM_WIDTH):
         """Get HTML representation of and item."""
-
-        display = html.escape(item_completion).replace(' ', '&nbsp;')
-        return COMPLETION_ITEM_TEMPLATE.format(completion=display,
-                                               type=item_type,
-                                               color=ima.MAIN_FG_COLOR,
-                                               icon_provider=icon_provider,
-                                               img_height=img_height,
-                                               img_width=img_width,
-                                               height=height,
-                                               width=width)
```

Comment:
Why not using a template here? It is far more readable and maintainable than using a list and joining its elements.

Reply:
`.format` is actually quite slow. If we were restricting ourselves to 3.7, then the new f-strings get precompiled into efficient code, but those aren't fast / available in older versions of Python.

Reply:
@goanpeca, what do you think?

Reply:
@metalogical, what about the old string replacement?

Reply:
I think old-style and new-style string formatting are comparable in performance -- they both need to make function calls and parse the format string every time you do the format operation.

f-strings are fast because Python compiles the syntax into efficient string processing logic. My approach is similarly faster than `.format` because it's similar to what the code would compile into, and doesn't require the parsing operation & function calls.

Reply:
Ok, I understand your reasoning now and see no other alternative for now.

I'd just like to ask you to add a comment above `parts = [` describing why we use here this approach instead of `.format` or string replacement.